components/accelerator/nvidia/gpu-counts/component.go

```diff
@@ -0,0 +1,290 @@
+	lastCheckResult *checkResult
+}
+
+func New(gpudInstance *components.GPUdInstance) (components.Component, error) {
+	cctx, ccancel := context.WithCancel(gpudInstance.RootCtx)
+	c := &component{
+		ctx:          cctx,
+		cancel:       ccancel,
+		nvmlInstance: gpudInstance.NVMLInstance,
+		getCountLspci: func(ctx context.Context) (int, error) {
+			devs, err := nvidiaquery.ListPCIGPUs(ctx)
+			if err != nil {
+				return 0, err
+			}
+			return len(devs), nil
+		},
+		getThresholdsFunc: GetDefaultExpectedGPUCounts,
+	}
+
+	return c, nil
+}
+
+func (c *component) Name() string { return Name }
+
+func (c *component) Tags() []string {
+	return []string{
+		"accelerator",
+		"gpu",
+		"nvidia",
+		Name,
+	}
+}
+
+func (c *component) IsSupported() bool {
+	if c.nvmlInstance == nil {
+		return false
+	}
+	return c.nvmlInstance.NVMLExists() && c.nvmlInstance.ProductName() != ""
+}
+
+func (c *component) Start() error {
+	go func() {
+		ticker := time.NewTicker(time.Minute)
+		defer ticker.Stop()
+
+		for {
+			_ = c.Check()
+
+			select {
+			case <-c.ctx.Done():
+				return
+			case <-ticker.C:
+			}
+		}
+	}()
+	return nil
+}
+
+func (c *component) LastHealthStates() apiv1.HealthStates {
+	c.lastMu.RLock()
+	lastCheckResult := c.lastCheckResult
+	c.lastMu.RUnlock()
+	return lastCheckResult.HealthStates()
+}
+
+func (c *component) Events(ctx context.Context, since time.Time) (apiv1.Events, error) {
+	return nil, nil
+}
+
+func (c *component) Close() error {
+	log.Logger.Debugw("closing component")
+
+	c.cancel()
+
+	return nil
+}
+
+func (c *component) Check() components.CheckResult {
+	log.Logger.Infow("checking nvidia gpu counts")
+
+	cr := &checkResult{
+		ts: time.Now().UTC(),
+	}
+
+	thresholds := c.getThresholdsFunc()
+	defer func() {
+		// nothing specified for this machine, gpud MUST skip the gpu count check
+		// do this in defer in order to check gpu count in gpud scan
+		// where it does not set the thresholds
+		if thresholds.IsZero() {
+			cr.reason = reasonThresholdNotSetSkipped
+			cr.health = apiv1.HealthStateTypeHealthy
+		}
+
+		c.lastMu.Lock()
+		c.lastCheckResult = cr
+		c.lastMu.Unlock()
+	}()
+
+	if c.nvmlInstance == nil {
+		cr.health = apiv1.HealthStateTypeHealthy
+		cr.reason = "NVIDIA NVML instance is nil"
+		return cr
+	}
+	if !c.nvmlInstance.NVMLExists() {
+		cr.health = apiv1.HealthStateTypeHealthy
+		cr.reason = "NVIDIA NVML library is not loaded"
+		return cr
+	}
+	if c.nvmlInstance.ProductName() == "" {
+		cr.health = apiv1.HealthStateTypeHealthy
+		cr.reason = "NVIDIA NVML is loaded but GPU is not detected (missing product name)"
+		return cr
+	}
+
+	cr.ProductName = c.nvmlInstance.ProductName()
+	cr.CountNVML = len(c.nvmlInstance.Devices())
+
+	if c.getCountLspci != nil {
+		cr.CountLspci, cr.err = c.getCountLspci(c.ctx)
+		if cr.err != nil {
+			cr.health = apiv1.HealthStateTypeUnhealthy
+			cr.reason = "error getting count of lspci"
+			log.Logger.Errorw(cr.reason, "error", cr.err)
+			return cr
+		}
+	}
```

Comment:
Is it necessary for this component to depend on lspci?

Reply:
Yes, let's keep them both for now. When we know this is stable, we can drop one (lspci).